lib/output.js

```diff
@@ -1185,6 +1200,21 @@ function OutputStream(options) {
         output.print(self.getValue());
     });
     DEFPRINT(AST_String, function(self, output){
+        var parent = output.parent(1);
```

Comment:
What if the parent or granparent node does not exist? Is an exception thrown or does the function `parent` return null?